BE/index.ts

```diff
@@ -1 +1,197 @@
-console.log("Hello, World!");
+import tasks from "./data/tasks.json";
+import listOfChallenges from "./data/listOfChallenges.json";
+
+import achievements from "./data/achievements.json";
+import defaultAchievements from "./data/defaultAchievements.json";
+import { checkAchievementsComplete } from "./utils/constants";
+import {
+  Challenge,
+  ActualAchievement,
+  ChallengeState,
+  Status,
+  StatusState,
+  Task,
+  ActualTask,
+  Achievement,
+  // @ts-ignore
+} from "./types";
+
+/**  return new shuffled array.
+ * @param {array} arr - array that should be shuffled
+ *
+ * */
+function shuffle(arr: any[]) {
+  const newArray = [...arr];
+  return newArray.sort(() => Math.random() - 0.5);
+}
+
+/**
+ * @param {array} arr - list of items that can be picked
+ * @param {number} count - a number of the items that should pick from the array.
+ */
+function pickRandomItems(arr: Array<any>, count: number) {
+  const shuffledArray = shuffle(arr);
+  if (shuffledArray.length > count) {
+    shuffledArray.length = count;
+  }
+  return shuffledArray;
+}
+
+/**
+ *
+ * @param arr - array of the data
+ * @param updatedDate - passed today for all items
+ * @param state - StatusState - can be Pending/Success/Failure, Pending - by default
+ *
+ * return {
+ *     [id]: { state:StatusState, updated: updatedDate }
+ *     ...}
+ */
+function createNewStatusState({
+  arr,
+  updatedDate,
+  state = StatusState.Pending,
+}: {
+  arr: Array<string>;
+  updatedDate: number;
+  state?: StatusState;
+}) {
+  return arr.reduce(
+    (acc, id) => ({ ...acc, [id]: { state, updated: updatedDate } }),
+    {},
+  );
+}
+
+/**
+ *
+ * @param challengeId - id of the current challenge
+ */
+export function getCurrentTask(challengeId: Challenge["id"]): ActualTask {
+  const challenge: Challenge = (<any>listOfChallenges)[challengeId];
```

Comment:
It would be better to avoid <any> usage
E.g we could cast json to proper types after import
And also  as challengeId is id from Challenge here could be more appropriate find the challenge by id 
`const challenge: Challenge = listOfChallenges.find(({ id }) => id === challengeId);`